blocks/ai-text/src/app/generate-text/model-selector.tsx

```diff
@@ -0,0 +1,322 @@
+      onClose={() => setOpen(false)}
+      value={model}
+      onChange={(event) => {
+        onModelChange(event.target.value);
+      }}
+      ref={(ref: HTMLSelectElement | null) => {
+        if (ref) {
+          setSelectRef(ref);
+        }
+      }}
+      renderValue={() => {
+        if (selectedModel) {
+          const { name } = selectedModel;
+          const { name: groupName, icon } = selectedGroup;
+          return (
+            <Typography
+              variant="regularTextLabels"
+              sx={{
+                display: "inline-flex",
+                gap: 1,
+                alignItems: "center",
+                fontSize: 15,
+                lineHeight: 1,
+                letterSpacing: -0.02,
+                color: ({ palette }) => palette.gray[50],
+              }}
+            >
+              Using
+              <Box
+                component="span"
+                sx={{
+                  display: "inline-flex",
+                  gap: 0.375,
+                  color: ({ palette }) => palette.gray[60],
+                }}
+              >
+                <Box sx={{ fontSize: 16 }}>{icon}</Box>
+                {groupName} {name}
+              </Box>
+            </Typography>
+          );
+        }
+
+        return null;
+      }}
+      inputProps={{
+        sx: {
+          paddingY: 0,
+          [`~.${outlinedInputClasses.notchedOutline}`]: {
+            border: "none !important",
+            boxShadow: "none !important",
+            background: "none !important",
+          },
+        },
+      }}
+      MenuProps={{
+        PaperProps: paperProps,
+        MenuListProps: {
+          sx: {
+            padding: 0,
+          },
+        },
+      }}
+      sx={{
+        [`.${inputBaseClasses.root}`]: {
+          boxShadow: "none !important",
+        },
+        ...(open
+          ? {
+              position: "relative",
+              zIndex: 9999,
+            }
+          : {}),
+      }}
+    >
+      <Box
+        ref={(ref: HTMLDivElement | null) => {
+          if (ref) {
+            setMenuRef(ref);
+          }
+        }}
+      >
+        {Object.values(MODELS_BY_GROUP).map(
+          ({ name: groupName, icon, models }) => (
+            <Box key={groupName}>
+              <Typography
+                sx={{
+                  fontWeight: 500,
+                  fontSize: 12,
+                  lineHeight: 1.3,
+                  letterSpacing: "0.03em",
+                  textTransform: "uppercase",
+                  color: ({ palette }) => palette.gray[50],
+                  mb: 1,
+                  padding: 2,
+                  paddingBottom: 0,
+                }}
+              >
+                {groupName}
+              </Typography>
+
+              <Stack>
+                {models.map(({ id, name, description }) => {
+                  const active = id === model;
+                  return (
+                    <MenuItem
```

Comment:
This doesn't seem to be being picked up by the `Select` as an option – there are these warnings in the console, even if this is given a `value`. I'm guessing that's why there is a custom `onClick` here, and the `onChange` on the `Select` is never triggered.

There are also no keyboard controls available for the `Select`, probably related. Maybe we can use [grouping](https://mui.com/material-ui/react-select/#grouping) but customise how they're displayed?

```
SelectInput.js?6fd7:406 MUI: 
You have provided an out-of-range value `gpt-3.5-turbo` for the select component.
Consider providing a value that matches one of the available options or ''.
The available values are "".
```